### ISSUE TITLE
Crash on AI turn

### ISSUE THREAD

**Issue by Rovdyr740** [vcmi/vcmi#3691]
**Describe the bug**
1. 
During the AI turn the sound of crash pops up and freezes the AI turn (oddly enough the animations continue). The crash pop-up is behind the game window, so i have to use tab to get it up. when i click "ok" then it goes back to main screen. this is the 2nd map i have tried this error on and it has been appearing after several hours of gaming.

Current Map: Fawn's Labyrinth MAS
other map with same issue: Neighbourhood2

2. HOTA maps from ver. 1.7 does not show up in maps when starting a game.

3. Crash file is nowhere to be found

4. Can't seem to find random map template (want to add some from HOTA RMG editor)

**Game logs**
Please attach game logs: `VCMI_client.txt`, `VCMI_server.txt` etc.

**To Reproduce**
added Save file, just press the hourglass to start AII turns for both maps.

**Expected behavior**
End AI turn and start my own

**Actual behavior**
Crash on AI turn

**Did it work earlier?**
Just started playing VCMI again

**Screenshots**
![vcmi CRASH](https://github.com/vcmi/vcmi/assets/161179294/df049a4e-b652-4c88-b5e7-d28590586af9)

**Version**
 - OS: Windows 11 Pro 64 bit
 - VCMI version 1.4.5

**Additional context**
I have tried reloading earlier but seems to run ito the samme problem either way.
I have tried using VCAI and nunkiller, no change
I've run refresh repository
[server_log.txt](https://github.com/vcmi/vcmi/files/14781307/server_log.txt)
[VCMI_Launcher_log.txt](https://github.com/vcmi/vcmi/files/14781308/VCMI_Launcher_log.txt)
[VCMI_Server_log.txt](https://github.com/vcmi/vcmi/files/14781309/VCMI_Server_log.txt)
[Save game both maps.zip](https://github.com/vcmi/vcmi/files/14781327/Save.game.both.maps.zip)
![active mods](https://github.com/vcmi/vcmi/assets/161179294/35efa592-b0da-4670-85a3-d704633fc0d5)

**Comment by nullkiller** [comment]
> I have tried using VCAI and nunkiller, no change
Expected, you can not change AI in save file, it will load original ones anyway.

Crash is 100% reproducible from save file?

**Comment by nullkiller** [comment]
Does not HOTA, WOG and Tides of War conflict with each other?

**Comment by Rovdyr740** [comment]
> > I have tried using VCAI and nunkiller, no change
> > Expected, you can not change AI in save file, it will load original ones anyway.
> 
> Crash is 100% reproducible from save file?

I've tried various things, so happen maybe 15/15 times I've tried to fix it, so I expect so yes

**Comment by Rovdyr740** [comment]
> Does not HOTA, WOG and Tides of War conflict with each other?

Since I'm 200 turns into the game then I would say no?

It's the only issue I've encountered and it seems to be related to the AI pathfinding every time... But I'm no expert

**Comment by IvanSavenko** [comment]
> Does not HOTA, WOG and Tides of War conflict with each other?

No, they don't.
However considering rather long list of mods it is possible that there are some mods that do conflict (and that were not marked as such)

**Comment by Rovdyr740** [comment]
> > Does not HOTA, WOG and Tides of War conflict with each other?
> 
> No, they don't. However considering rather long list of mods it is possible that there are some mods that do conflict (and that were not marked as such)

Good point, maybe boost AI and the "native" vcmi ai? But it is just very late that it manifests...

**Comment by nullkiller** [comment]
I will try to download all of them and reproduce. At least all information needed seams to be present.

**Comment by Rovdyr740** [comment]
> I will try to download all of them and reproduce. At least all information needed seams to be present.

Let me know if I can help... I can send the maps when I get home in the evening if needed

**Comment by nullkiller** [comment]
Would be nice if you can try to download https://github.com/vcmi/vcmi/actions/runs/8379749785/artifacts/1347688896 (develop a few days ago)
and see if you can generate a crash dump. Or alternative windows build from 1.4.5
Crash dump should be generated near exe file.

**Comment by Rovdyr740** [comment]
> Would be nice if you can try to download https://github.com/vcmi/vcmi/actions/runs/8379749785/artifacts/1347688896 (develop a few days ago) and see if you can generate a crash dump. Or alternative windows build from 1.4.5 Crash dump should be generated near exe file.

That's the thing, the crash dump does not generate, I've been through all folders without luck.

I'll try the download in the evening when I get home.

**Comment by Povelitelh3** [comment]
I tried this map. 
What size of your RAM? 
because ![изображение](https://github.com/vcmi/vcmi/assets/36762538/ce16e12f-4a56-45d1-a55e-484df16cb06d)

**Comment by Rovdyr740** [comment]
> I tried this map. What size of your RAM? because ![изображение](https://private-user-images.githubusercontent.com/36762538/317776142-ce16e12f-4a56-45d1-a55e-484df16cb06d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2MzkzMDIsIm5iZiI6MTcxMTYzOTAwMiwicGF0aCI6Ii8zNjc2MjUzOC8zMTc3NzYxNDItY2UxNmUxMmYtNGE1Ni00NWQxLWE1NWUtNDg0ZGYxNmNiMDZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI4VDE1MTY0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0NDE1MjQ3MWVmZTQwYjg5ODQ2ZDZlNzlkMzVmMzliZWY2MzI0YmVhMzFhMTJmNmQ0YzM5ZGNmNTQyYzhiOWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Np1nQerDPKuasDlcLZnfxdiEIXuPd_O_mvhQmUofFQ4)

CPU: i7 10700kf
Gpu: rtx 4070
M. 2 drive
Ram: 32gn (2x16gb 3733 mz)

**Comment by Rovdyr740** [comment]
> Would be nice if you can try to download https://github.com/vcmi/vcmi/actions/runs/8379749785/artifacts/1347688896 (develop a few days ago) and see if you can generate a crash dump. Or alternative windows build from 1.4.5 Crash dump should be generated near exe file.

Crashes when i try too load the save file with the linked build

**Comment by Rovdyr740** [comment]
[The two maps.zip](https://github.com/vcmi/vcmi/files/14795610/The.two.maps.zip)
 
Added the two maps here for you to test

**Comment by nullkiller** [comment]
Some attempts to run this map caused internal memory corruptions. Will investigate further

**Comment by nullkiller** [comment]
Looks like CMap object contains nullptr quests. As far as I know this should not happen. Not sure what to do next

**Comment by DjWarmonger** [comment]
Let me look into it, I just started refactoring CMap today.

**Comment by nullkiller** [comment]
At least with close setup of mods I managed to narrow down some interesting effect. Dismissing a hero who starts in water causes crash on AI turn later.

**Comment by nullkiller** [comment]
Boat looks being present in starting tile while when we dismiss the owning hero it gets destroyed but a reference is still present in tile

**Comment by Rovdyr740** [comment]
> Boat looks being present in starting tile while when we dismiss the owning hero it gets destroyed but a reference is still present in tile

Now that you mention it then it fits with what happened on the neighbourhood map.. The AI was battling on boats, one lost and shortly after the crash occurred. It is definitely also a possibility on the labyrinth map, haven't removed the fow to verify.

**Comment by nullkiller** [comment]
Looks like I managed to fix boat issue

**Comment by Rovdyr740** [comment]
> Looks like I managed to fix boat issue

Awesome!

Thank you very  much for spending the time!

When will it be released?

And do you have a donation system setup, really enjoy your work and after this great error handling, then I'd like to contribute a little. I haven't got much, but little is better than nothing...

**Comment by Rovdyr740** [comment]
@nullkiller How doo I apply the fix? Just redownloaded newest launcher and updated repos. and all new mod updates but crash still occurs is there a file i should replace or some doc i need too edit?

**Comment by nullkiller** [comment]
Unfortunately by starting a new game on this map. As this is initial game load issue.

**Comment by nullkiller** [comment]
Also you can try to kill all these heroes on water and save and load game and see if it will crash or prevent boats from being destroyed somehow.
